File: solutions/julia/raindrops/2/raindrops.jl
function raindrops(number::Int)::String
    result = ""

    number % 3 == 0 && (result *= "Pling")

    number % 5 == 0 && (result *= "Plang")

    number % 7 == 0 && (result *= "Plong")

    result == "" ? string(number) : result
end

File: solutions/kotlin/raindrops/1/src/main/kotlin/Raindrops.kt
class Raindrops {
    companion object {
        fun convert(input: Int): String {

            val output = mutableListOf<String>()
            val divisors = hashMapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")

            divisors.toSortedMap().forEach {
                if (input.rem(it.key) == 0) {
                    output.add(it.value)
                }
            }

            if (output.isNotEmpty()) {
                return output.joinToString(separator = "")
            }

            return input.toString()
        }
    }
}